### ISSUE TITLE
Pylance incorrectly marks decorator instance methods of a class for not providing the `self` parameter.

### ISSUE THREAD

**Issue by DeszkaCodes** [microsoft/pyright#10262]
Pylance incorrectly marks decorator instance methods of a class for not providing the `self` parameter.

## Environment data

-   Pylance version: 2025.4.1
-   OS and version: Microsoft Windows 10 Pro, 10.0.19045 Build 19045
-   Python version (& distribution if applicable, e.g. Anaconda): 3.13.2 (via `winget`)

## Code Snippet

```python
from collections.abc import Callable
import functools
from typing import ParamSpec, TypeVar


P = ParamSpec("P")
R = TypeVar("R")

def log(f: Callable[P, R]):
    @functools.wraps(f)
    def wrapper(*args:P.args, **kwargs:P.kwargs) -> R:
        fname: str = f.__name__
        print(fname, "entered")
        val: R = f(*args, **kwargs)
        print(fname, "exited:", val)
        return val
    
    return wrapper


class Calculator:
    @log
    def add(self, a: int, b:int)->int:
        return a + b
    
calc = Calculator()
calc.add(2, 3)
```

## Repro Steps

1. Create a correctly type-hinted decorator
2. Create a class with an instance method
3. Add the decorator to the instance method
4. Create an instance of the class and call the method
5. See the error Pylance shows

## Expected behavior

Pylance infers `self` correctly and no error is shown.

## Actual behavior

Pylance cannot recognize that `self` will be provided by the interpreter and expects the user to provide it (which would cause an error)

Pylance error:
```
Argument missing for parameter "b"Pylance[reportCallIssue](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportCallIssue)
(method) add: _Wrapped[(self: Calculator, a: int, b: int), int, (self: Calculator, a: int, b: int), int]
```

## Logs

[Python Language Server.log](https://github.com/user-attachments/files/19617036/Python.Language.Server.log)
